diff --git a/src/main/java/org/libvirt/SchedParameter.java b/src/main/java/org/libvirt/SchedParameter.java
index ab988e9..d66c5f2 100644
--- a/src/main/java/org/libvirt/SchedParameter.java
+++ b/src/main/java/org/libvirt/SchedParameter.java
@@ -24,13 +24,13 @@ public abstract class SchedParameter {
                     returnValue = new SchedIntParameter(vParam.value.i);
                     break;
                 case (2):
-                    returnValue = new SchedUintParameter(vParam.value.ui);
+                    returnValue = new SchedUintParameter(vParam.value.i);
                     break;
                 case (3):
                     returnValue = new SchedLongParameter(vParam.value.l);
                     break;
                 case (4):
-                    returnValue = new SchedUlongParameter(vParam.value.ul);
+                    returnValue = new SchedUlongParameter(vParam.value.l);
                     break;
                 case (5):
                     returnValue = new SchedDoubleParameter(vParam.value.d);
@@ -55,7 +55,7 @@ public abstract class SchedParameter {
                 returnValue.value.setType(int.class);
                 break;
             case (2):
-                returnValue.value.ui = ((SchedUintParameter) param).value;
+                returnValue.value.i = ((SchedUintParameter) param).value;
                 returnValue.value.setType(int.class);
                 break;
             case (3):
@@ -63,7 +63,7 @@ public abstract class SchedParameter {
                 returnValue.value.setType(long.class);
                 break;
             case (4):
-                returnValue.value.ul = ((SchedUlongParameter) param).value;
+                returnValue.value.l = ((SchedUlongParameter) param).value;
                 returnValue.value.setType(long.class);
                 break;
             case (5):
diff --git a/src/main/java/org/libvirt/jna/virSchedParameterValue.java b/src/main/java/org/libvirt/jna/virSchedParameterValue.java
index 903362c..8d036e4 100644
--- a/src/main/java/org/libvirt/jna/virSchedParameterValue.java
+++ b/src/main/java/org/libvirt/jna/virSchedParameterValue.java
@@ -7,9 +7,7 @@ import com.sun.jna.Union;
  */
 public class virSchedParameterValue extends Union {
     public int i; /* data for integer case */
-    public int ui; /* data for unsigned integer case */
     public long l; /* data for long long integer case */
-    public long ul; /* data for unsigned long long integer case */
     public double d; /* data for double case */
     public byte b; /* data for char case */
 }
